src/createXcmTypes/util/getSystemChainAssetId.ts

```diff
@@ -0,0 +1,99 @@
+// Copyright 2023 Parity Technologies (UK) Ltd.
+
+import { ApiPromise } from '@polkadot/api';
+
+import { SYSTEM_PARACHAINS_IDS } from '../../consts';
+import { BaseError } from '../../errors';
+import { Registry } from '../../registry';
+import { foreignAssetMultiLocationIsInRegistry } from './foreignAssetMultiLocationIsInRegistry';
+import { getChainIdBySpecName } from './getChainIdBySpecName';
+import { isValidForeignAssetMultiLocation } from './isValidForeignAssetMultiLocation';
+/**
+ * Returns the correct asset id for a valid system chain token symbol
+ * integer id or foreign asset multilocation
+ * errors if given an invalid symbol, integer id or multilocation that is
+ * not registered or found in the chains state
+ *
+ * @param tokenSymbol string
+ * @param specName string
+ */
+export const getSystemChainAssetId = async (
+	asset: string,
+	specName: string,
+	_api: ApiPromise,
+	transferForeignAssets?: boolean | undefined
+): Promise<string> => {
+	let assetId = '';
+	const newRegistry = new Registry(specName, {});
+	const systemChainId = getChainIdBySpecName(newRegistry, specName);
+
+	if (!SYSTEM_PARACHAINS_IDS.includes(systemChainId)) {
+		throw new BaseError(
+			`specName ${specName} did not match a valid system chain ID. Found ID ${systemChainId}`
+		);
+	}
+
+	if (transferForeignAssets && systemChainId === '1000') {
+		// determine if we
+		const multiLocationIsInRegistry = foreignAssetMultiLocationIsInRegistry(
+			asset,
+			newRegistry,
+			_api
+		);
+
+		if (multiLocationIsInRegistry) {
+			assetId = asset;
```

Comment:
Does this assume that the assetId is a `multiLocation` as a string?

Reply:
yes, the `isForeignAssetsTransfer` boolean is only true if the assetId is a `MultiLocation` as a string.

Reply:
Ahh gotcha! nice!